diff --git a/js/range-slider-widget.js b/js/range-slider-widget.js
index d6809dd..e5588fe 100644
--- a/js/range-slider-widget.js
+++ b/js/range-slider-widget.js
@@ -1,29 +1,32 @@
 (function ($, window, Drupal) {
+
   Drupal.behaviors.rangeSliderWidget = {
     attach: function attach() {
-      $(".range-slider-widget-field").each(function(){
-        var $element = $(this);
-        var min = parseFloat($element.attr("min")) || 0;
-        var max = parseFloat($element.attr("max")) || (min+100);
-        var step = parseFloat($element.attr("step")) || 0.1;
-        var value = parseFloat($element.val()) || min;
-        var $container = $element.parent();
-        var $slider = $container && $container.find('.range-slider-widget');
+      var $fields = document.querySelectorAll('.range-slider-widget-field');
+      for (var i = 0; i < $fields.length; ++i) {
+        var $field = $fields[i];
+        var $min = parseFloat($field.getAttribute("min")) || 0;
+        var $max = parseFloat($field.getAttribute("max")) || (min+100);
+        var $step = parseFloat($field.getAttribute("step")) || 0.1;
+        var $value = parseFloat($field.getAttribute("value")) || 0.1;
+        var $container = $field.parentNode;
+        var $slider = $container.querySelector('.range-slider-widget');
 
-        if($slider && $slider.length > 0) {
-          $slider.slider({
-            step: step,
-            min: min,
-            max: max,
-            value: value,
-            slide: function( event, ui ) {
-              $element.val( ui.value );
-            }
+        if ($slider && $slider.length > 0) {
+          noUiSlider.create($slider, {
+            start: $value,
+            range: {
+              'min': $min,
+              'max': $max
+            },
+            step: $step
+          });
+          $slider.noUiSlider.on('update', function (values, handle) {
+            $field.setAttribute('value', values[handle]);
           });
         }
-
-      });
-
+      }
     }
   };
+
 })(jQuery, window, Drupal);
diff --git a/range_slider_widget.libraries.yml b/range_slider_widget.libraries.yml
index ecabc48..e2f83de 100644
--- a/range_slider_widget.libraries.yml
+++ b/range_slider_widget.libraries.yml
@@ -1,10 +1,24 @@
+nouislider:
+  remote: https://github.com/leongersen/noUiSlider
+  license:
+    name: MIT
+    url: https://github.com/leongersen/noUiSlider/blob/master/LICENSE.md
+    gpl-compatible: true
+  version: 14.6.1
+  css:
+    /libraries/nouislider/distribute/nouislider.min.css: { weight: -1, minified: true }
+  js:
+    /libraries/nouislider/distribute/nouislider.min.js: { weight: -1, minified: true }
+
+
 range_slider_widget:
   version: 1.x
   css:
     theme:
       css/range-slider-widget.css: {}
   js:
-    js/range-slider-widget.js: {}
+
+    js/range-slider-widget.js: { weight: 0 }
   dependencies:
     - core/jquery
-    - core/jquery.ui.slider
+    - range_slider_widget/nouislider
