diff --git a/js/range-slider-widget.js b/js/range-slider-widget.js
index d6809dd..adf749a 100644
--- a/js/range-slider-widget.js
+++ b/js/range-slider-widget.js
@@ -1,29 +1,32 @@
 (function ($, window, Drupal) {
+
   Drupal.behaviors.rangeSliderWidget = {
-    attach: function attach() {
-      $(".range-slider-widget-field").each(function(){
-        var $element = $(this);
-        var min = parseFloat($element.attr("min")) || 0;
-        var max = parseFloat($element.attr("max")) || (min+100);
-        var step = parseFloat($element.attr("step")) || 0.1;
-        var value = parseFloat($element.val()) || min;
-        var $container = $element.parent();
-        var $slider = $container && $container.find('.range-slider-widget');
+    attach: function attach(context) {
+      var $fields = $('.range-slider-widget-field', context).once('rangeSliderWidget').each(function() {
+        var $field = $(this);
+        var $min = parseFloat($field.attr('min')) || 0;
+        var $max = parseFloat($field.attr('max')) || (min+100);
+        var $step = parseFloat($field.attr('step')) || 0.1;
+        var $value = parseFloat($field.val()) || min;
+        var $container = $field.parent();
+        var $slider = $container.find('.range-slider-widget')[0];
+
+        if ($slider !== "undefined") {
+          noUiSlider.create($slider, {
+            start: [$value],
+            range: {
+              'min': [$min],
+              'max': [$max]
+            },
+            step: $step
+          }, true);
 
-        if($slider && $slider.length > 0) {
-          $slider.slider({
-            step: step,
-            min: min,
-            max: max,
-            value: value,
-            slide: function( event, ui ) {
-              $element.val( ui.value );
-            }
+          $slider.noUiSlider.on("update", function(values, handle) {
+            $field.val(values[handle]);
           });
         }
-
       });
-
     }
   };
+
 })(jQuery, window, Drupal);
diff --git a/range_slider_widget.libraries.yml b/range_slider_widget.libraries.yml
index ecabc48..1423e58 100644
--- a/range_slider_widget.libraries.yml
+++ b/range_slider_widget.libraries.yml
@@ -1,10 +1,23 @@
+nouislider:
+  remote: https://github.com/leongersen/noUiSlider
+  license:
+    name: MIT
+    url: https://github.com/leongersen/noUiSlider/blob/master/LICENSE.md
+    gpl-compatible: true
+  version: 14.6.1
+  css:
+    /libraries/nouislider/distribute/nouislider.min.css: { minified: true }
+  js:
+    /libraries/nouislider/distribute/nouislider.min.js: { minified: true }
+
 range_slider_widget:
   version: 1.x
   css:
     theme:
       css/range-slider-widget.css: {}
   js:
+
     js/range-slider-widget.js: {}
   dependencies:
     - core/jquery
-    - core/jquery.ui.slider
+    - range_slider_widget/nouislider
